site-builder/src/site/resource.rs

```diff
@@ -56,14 +52,59 @@ impl TryFrom<&SuiMoveStruct> for ResourceInfo {
 
         Ok(Self {
             path,
-            content_type,
-            content_encoding,
+            headers,
             blob_id,
             blob_hash,
         })
     }
 }
 
+#[derive(serde::Serialize, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct HttpHeader {
+    pub name: String,
+    pub value: String,
+}
+
+#[derive(serde::Serialize, Clone, Debug, PartialEq, PartialOrd, Ord, Eq)]
+pub struct HttpHeaders(pub Vec<HttpHeader>);
+
+impl HttpHeaders {
+    pub fn new() -> Self {
+        HttpHeaders(Vec::new())
+    }
+}
+
+impl From<HashMap<String, String>> for HttpHeaders {
+    fn from(values: HashMap<String, String>) -> HttpHeaders {
+        let mut headers = Vec::new();
+        for value in values {
+            headers.push(HttpHeader {
+                name: value.0,
+                value: value.1,
+            });
+        }
+        HttpHeaders(headers)
```

Comment:
Nit: slightly more idiomatic:
```suggestion
    fn from(values: HashMap<String, String>) -> Self {
        Self(
            values
                .into_iter()
                .map(|(name, value)| HttpHeader { name, value })
                .collect(),
        )
    }
```